plugin/aws/root.go

```diff
@@ -60,10 +61,25 @@ func exists(path string) error {
 }
 
 // Init for root
-func (r *Root) Init() error {
+func (r *Root) Init(cfg map[string]interface{}) error {
 	r.EntryBase = plugin.NewEntry("aws")
 	r.SetTTLOf(plugin.ListOp, 1*time.Minute)
 
+	if profsI, ok := cfg["profiles"]; ok {
```

Comment:
Why not have a `Config` struct that you can marshal this into?

Reply:
Didn't think of it.

Reply:
Also it gets messy unless I remarshal to JSON and unmarshal again.

Reply:
Hm, maybe we could extend `Root` with a `ConfigStruct` method to return a base config object so we can handle the marshalling for them? Then plugin authors just have to type-assert the passed-in struct.

So something like `Init(config interface{})`, where `config` is their specified struct.

Reply:
I think it's not much more complicated than marshal/unmarshal, and provides clearer errors.

Reply:
Either approach is OK with me.

Reply:
The ConfigStruct concept moves the conversion around but we'd still have to handle it. Because we're not originally unmarshaling JSON, we're calling `viper.GetStringMap()`. So an improvement I'm tempted to defer for now.